addons/website_sale_stock_wishlist/controllers/main.py

```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+from odoo import http
+from odoo.http import request
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSaleStockWishlist(WebsiteSale):
+    @http.route(['/shop/wishlist/notify/<model("product.wishlist"):wish>'], type='json', auth="public", website=True)
+    def notify_stock(self, wish, **kw):
+        if not request.website.is_public_user():
+            wish['stock_notification'] = not wish.stock_notification
+        return True
```

Comment:
Even though UI might be a toggle operation, it feels safer to provide the new value as a parameter.
If several tabs are opened, one of them could be out of sync and be misleading.